pkg/controller/daemonset/daemonset_controller.go

```diff
@@ -127,7 +134,11 @@ func Add(mgr manager.Manager) error {
 	if !utildiscovery.DiscoverGVK(controllerKind) {
 		return nil
 	}
-
+	if !utildiscovery.DiscoverGVK(appsv1alpha1.SchemeGroupVersion.WithKind("ImagePullJob")) ||
+		!utilfeature.DefaultFeatureGate.Enabled(features.KruiseDaemon) ||
+		!utilfeature.DefaultFeatureGate.Enabled(features.PreDownloadImageForInPlaceUpdate) {
```

Comment:
plz add a separate feature gate for PreDownloadImageForDaemonSet that is disabled by default, predownload for ads is still lack some feature e.g. updateReadyPods check before predownload